core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/source/sql/asterixdb/AsterixDBSourceOpDesc.scala

```diff
@@ -61,7 +104,11 @@ class AsterixDBSourceOpDesc extends SQLSourceOpDesc {
           batchByColumn,
           min,
           max,
-          interval
+          interval,
+          geoLocation,
+          geoAttributes,
+          searchByColumnForRegex,
+          regex
```

Comment:
for keyword search, we have `search`, `searchByColumn` and `keywords`. I see now we extend search to be `FilterConditions` in general. 

Thus, I wonder if this is a good idea:
1. For each filter conditions, including keyword search, define them to be `keywordSearch: Boolean`, `keywordSearchByColumn: String`, and `keywordSearchCondition: String`.
2. pass all three fields for each filter condition to exec. 
3. use boolean in exec to check:
```
if (keywordSearch) {
 addKeywordSearch(queryBuilder)
}

if (regexSearch) {
 addRegexSearch(queryBuilder)
}

if (...){
}
```
My point is to make all filter conditions symmetric.

Reply:
done